refactor: Split pucoti in multiple files

ddorn opened the pull request:
Would love to have a review from you, @niajane !

niajane commented on src/app.py:
```diff
@@ -0,0 +1,444 @@
+from src import platforms
+
+
+def play(sound):
```
hmm maybe utils doesn't feel quite right

niajane commented:
This bit is personal style, but if you were wondering what I’d do, I like lots of small files so I would probably do something like the following:
1. App.py is only the app and main functions
2. Put the Scene, Purpose and Countdown Callback classes in a different file - I might even put them each in their own file :o
3. Split platforms.py into window_manager and os_specific_window_handler or something, with the os_specific_… file containing a function that only handles the os-specific things.
4. Possibly split utils, one way could be into time_formatter with all the functions that format time and utils for the remaining

...maybe. sometimes hard to tell what makes sense before you try it!

I guess a lot of the reasons for splitting things out so much is easier testing, so that doesn’t apply here :-)

niajane commented:
yay for refactor 🥳

niajane commented:
This bit is personal style, but if you were wondering what I’d do, I like lots of small files so I would probably do something like the following:

    App.py is only the app and main functions
    Put the Scene, Purpose and Countdown Callback classes in a different file - I might even put them each in their own file :o
    Split platforms.py into window_manager and os_specific_window_handler or something, with the os_specific_… file containing a function that only handles the os-specific things.
    Possibly split utils, one way could be into time_formatter with all the functions that format time and utils for the remaining

...maybe. sometimes hard to tell what makes sense before you try it!

I guess a lot of the reasons for splitting things out so much is easier testing, so that doesn’t apply here :-)

ddorn commented:
Thank you for your review! I enjoyed having this feedback on my code.

I implemented most of your feedback:
- Split `utils.py` into `time_utils.py` and `pygame_utils.py`. I think it fits nicely, except for `shorten()` but didn't want to split files too much either. 
- I put `adjust_window_size` in pygame_utils.py as it fits well, so platforms.py is only platform-specific stuff.
- I haven't split Purpose, Scene and Callback (yet). I think I would be happy to remove the Purpose and Callback from the main file and put them into a single one, but not Scene, as Scene is stringly typed (for the names of the different regions). My problem is that I don't know how to name this file. I'm hestating between `utils.py`, and `core.py`, both of which seem not so great? Or maybe I should accept to split in more files, even if I find the code harder to navigate.
- Also, there is currently no testing, but it's more that I haven't put the time to add some, I think it would be nice, for some parts.

niajane commented:
Nice!

Yes I guess that having Purpose and Callback in their own files makes them much easier to name. 
I'm struggling to think of one
